wgpu-hal/src/dx12/adapter.rs

```diff
@@ -103,8 +106,36 @@ impl super::Adapter {
             } else {
                 wgt::DeviceType::DiscreteGpu
             },
+            mpo: false,
         };
 
+        unsafe {
+            if let Ok(adapter1) = adapter.cast::<dxgi::IDXGIAdapter1>().into_result() {
+                for i in 0.. {
+                    let mut output = native::WeakPtr::<dxgi::IDXGIOutput>::null();
+                    match adapter1
+                        .EnumOutputs(i, output.mut_void() as _)
```

Comment:
based on this code, `mpo` would be a property of the pair (adapter, surface), not just the adapter alone (i.e. surface is presented via one of these outputs). So perhaps we should move `mpo` flag to the `SurfaceCapabilities`?

Reply:
[IDXGIAdapter::EnumOutputs](https://docs.microsoft.com/en-us/windows/win32/api/dxgi/nf-dxgi-idxgiadapter-enumoutputs) says that the `EnumOutputs` method enumerate adapter (video card) outputs, which I don't think has anything to do with the surface

Reply:
I'm confused, [vkCreateDisplayPlaneSurfaceKHR](https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkCreateDisplayPlaneSurfaceKHR.html) and [DXGI_SWAP_CHAIN_FLAG_FOREGROUND_LAYER](https://docs.microsoft.com/en-us/windows/win32/api/dxgi/ne-dxgi-dxgi_swap_chain_flag) seem to explicitly create surfaces on MPOs, but even without them the window compositor will present surfaces as MPOs when possible.